Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabLoader.h

```diff
@@ -141,6 +153,9 @@ namespace AzToolsFramework
                 TemplateId targetTemplateId,
                 AZStd::unordered_set<AZ::IO::Path>& progressedFilePathsSet);
 
+            bool SanitizeLoadedTemplate(PrefabDomReference loadedTemplateDom);
+            bool SanitizeSavingTemplate(PrefabDomReference savingTemplateDom);
```

Comment:
These methods really should have comments explaining what they're doing, and they're questionably named.  "Sanitize" implies "cleaning up", but there's no data validation or replacement happening here.  It would be more accurate to name them "DecompressTemplate" and "CompressTemplate", or even better, "ExpandDefaultValues" and "RemoveDefaultValues".  That would also help make these methods a bit more generically useful than tying them to the singular context of the act of loading and saving.  

If they're named "Sanitize" because they're intended to be an entry point for all data load/save manipulations, then perhaps they should be thin pass-through calls that would currently just call Expand/Remove methods and have comments saying that other steps could be added.  Otherwise, we're just encouraging multiple different operations to get added directly into a single generic catch-all function over time.

For the generic catch-all names, I'd also suggest something _less_ symmetrical because we won't actually need the same set of operations over time.  Loading should be more complex because it's handling arbitrary user-created content and actually *does* require complex data validation and cleansing.  Saving should be more straightforward because it's theoretically starting from correctly-formed data and shouldn't require any validation / cleansing, and presumably far less manipulation.  Maybe something like:
SanitizeLoadedTemplate(...)  // clean up any invalid user data and prepare template for runtime use
PrepareTemplateForSaving(...) // manipulate data into the desired save format

Reply:
I agree with the sentiment. I will add comments, but since as mentioned above this is intended to grow over time I will leave the current names. @sconel @srikappa-amzn make sure to keep this in mind when moving on with further refactorings :)